src/robocop/config.py

```diff
@@ -680,12 +800,18 @@ def overwrite_from_config(self, overwrite_config: Config | None) -> None:
                 "linter",
                 "formatter",
                 "file_filters",
+                "cache",
                 "config_source",
             ):  # TODO Use field metadata
                 continue
             value = getattr(overwrite_config, config_field.name)
             if value:
                 setattr(self, config_field.name, value)
+        # Handle cache config - CLI cache settings override file config
+        if overwrite_config.cache is not None and (
+            overwrite_config.cache.enabled is False or overwrite_config.cache.cache_dir is not None
```

Comment:
Is there a reason why we would like to overwrite config only when enabled is False or cache_dir is set? I know that the first though would be 'no point overwriting if it's the same as default' but what about this case:

Configuration file
```
[tool.robocop]
cache = false
```

CLI
```
robocop check --cache
```

The configuration file explictly says 'disable the cache' but we don't want to enable it in this run, so we call --no-cache.